hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/utils/factory/CollectSinkTableFactory.java

```diff
@@ -61,7 +61,7 @@ public class CollectSinkTableFactory implements DynamicTableSinkFactory {
   public static final String FACTORY_ID = "collect";
 
   // global results to collect and query
-  public static final Map<Integer, List<Row>> RESULT = new HashMap<>();
+  public static final Map<Integer, List<Row>> RESULT = new ConcurrentHashMap<>();
```

Comment:
why this change?

Reply:
Found a flaky test in the ci: https://github.com/apache/hudi/pull/13590#issuecomment-3610759021